File: todo.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

struct Task {
    string title;
    bool done;
};

vector<Task> loadTasks() {
    vector<Task> tasks;
    ifstream file("tasks.txt");
    string line;
    while (getline(file, line)) {
        bool done = line.rfind("[x]", 0) == 0;
        string title = line.substr(3);
        tasks.push_back({title, done});
    }
    file.close();
    return tasks;
}

void saveTasks(const vector<Task>& tasks) {
    ofstream file("tasks.txt");
    for (auto &t : tasks) {
        file << (t.done ? "[x]" : "[ ]") << t.title << "\n";
    }
    file.close();
}

void listTasks(const vector<Task>& tasks) {
    cout << "\nYour To-Do List:\n";
    if (tasks.empty()) {
        cout << "No tasks yet!\n";
        return;
    }
    for (size_t i = 0; i < tasks.size(); i++) {
        cout << i + 1 << ". " << (tasks[i].done ? "[✔]" : "[ ]") 
             << " " << tasks[i].title << "\n";
    }
}

int main() {
    vector<Task> tasks = loadTasks();
    int choice;

    do {
        cout << "\n==== To-Do List ====\n";
        cout << "1. View Tasks\n";
        cout << "2. Add Task\n";
        cout << "3. Mark Task Done\n";
        cout << "4. Exit\n";
        cout << "Choose: ";
        cin >> choice;
        cin.ignore();

        if (choice == 1) {
            listTasks(tasks);
        } 
        else if (choice == 2) {
            cout << "Enter task: ";
            string title;
            getline(cin, title);
            tasks.push_back({title, false});
            saveTasks(tasks);
            cout << "Added!\n";
        } 
        else if (choice == 3) {
            listTasks(tasks);
            cout << "Enter task number: ";
            int num;
            cin >> num;
            if (num > 0 && num <= tasks.size()) {
                tasks[num - 1].done = true;
                saveTasks(tasks);
                cout << "Marked as done!\n";
            } else {
                cout << "Invalid number.\n";
            }
        }
    } while (choice != 4);

    cout << "Goodbye!\n";
    return 0;
}

File: todo.py
import json
import os

FILE = "tasks.json"

def load_tasks():
    if os.path.exists(FILE):
        with open(FILE, "r") as f:
            return json.load(f)
    return []

def save_tasks(tasks):
    with open(FILE, "w") as f:
        json.dump(tasks, f, indent=2)

def list_tasks(tasks):
    if not tasks:
        print("\nNo tasks yet!")
        return
    print("\nYour To-Do List:")
    for i, t in enumerate(tasks, 1):
        status = "✔" if t["done"] else "✗"
        print(f"{i}. [{status}] {t['title']}")

def add_task(tasks):
    title = input("Enter new task: ")
    tasks.append({"title": title, "done": False})
    save_tasks(tasks)
    print("Task added!")

def mark_done(tasks):
    list_tasks(tasks)
    try:
        n = int(input("Task number to mark done: "))
        tasks[n-1]["done"] = True
        save_tasks(tasks)
        print("Marked as done!")
    except:
        print("Invalid number!")

def delete_task(tasks):
    list_tasks(tasks)
    try:
        n = int(input("Task number to delete: "))
        tasks.pop(n-1)
        save_tasks(tasks)
        print("Task deleted!")
    except:
        print("Invalid number!")

def main():
    tasks = load_tasks()
    while True:
        print("\n==== To-Do Menu ====")
        print("1. View Tasks")
        print("2. Add Task")
        print("3. Mark Task Done")
        print("4. Delete Task")
        print("5. Exit")
        choice = input("Choose: ")

        if choice == "1":
            list_tasks(tasks)
        elif choice == "2":
            add_task(tasks)
        elif choice == "3":
            mark_done(tasks)
        elif choice == "4":
            delete_task(tasks)
        elif choice == "5":
            print("Goodbye!")
            break
        else:
            print("Invalid option!")

if __name__ == "__main__":
    main()
